recipes/stb/all/conanfile.py

```diff
@@ -0,0 +1,29 @@
+from conans import ConanFile, tools
+import os
+
+class StbConan(ConanFile):
+    name = "stb"
+    description = "single-file public domain libraries for C/C++"
+    topics = ("conan", "stb", "single-file")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/nothings/stb"
+    license = ("Public domain", "MIT")
+    no_copy_source = True
+    _source_subfolder = "source_subfolder"
+    _commits = {
+        "20200203": "0224a44a10564a214595797b4c88323f79a5f934",
+    }
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self._commits[self.version]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        self.copy("*.h", src=self._source_subfolder, dst="include")
+        tools.rmdir(os.path.join(self.package_folder, "include", "tests"))
+
+    def package_info(self):
+        self.info.header_only()
```

Comment:
This should be in `package_id`

Reply:
@uilianries @danimtb Shouldn't `self.info` be `None` in all functions except `package_id()`?

Reply:
good question. I believe it could be easily spotted either in conan itself, or in hook. this is relatively common mistake to misuse package_id and package_info.

Reply:
yes, this is going to be addressed in the following Conan releases. I wouldn't spend time on creating a hook